### PR TITLE
fix(lease.cr): return nil when ttl root is missing

### PR DESCRIPTION
**Pull request by dukeraphaelng**
- Resolves https://github.com/place-labs/hound-dog/pull/6#issue-584387452

### CODE REVIEW

File: src/etcd/lease.cr [hunk 1]
```diff
@@ -15,8 +15,11 @@ class Etcd::Lease
 
   # Requests persistence of lease.
   # Must be invoked periodically to avoid key loss.
-  def keep_alive(id : Int64)
+  def keep_alive(id : Int64) : Int64?
     Model::KeepAlive.from_json(client.api.post("/lease/keepalive", {ID: id}).body).result
+  rescue e : Exception
```
Comment (caspiano) [inline src/etcd/lease.cr:20]: This should be a JSON parse exception

File: src/etcd/lease.cr [hunk 2]
```diff
@@ -15,8 +15,11 @@ class Etcd::Lease
...
+    raise e unless e.message == "JSON key not found: TTL"
+    nil
```
Comment (caspiano) [inline src/etcd/lease.cr:22]: Are you sure that a missing TTL key indicates a lost lease?

File: spec/helper.cr [hunk 3]
```diff
@@ -11,6 +11,6 @@ end
 Spec.before_each do
   begin
     Etcd.from_env.kv.delete_prefix TEST_PREFIX
-  rescue e
+  rescue
```
Comment (caspiano) [inline spec/helper.cr:14]: Nice little one-liner that's helpful at other times too! Allows you to just have a default return value if you don't really care about the exception.

```suggestion
  Etcd.from_env.kv.delete_prefix TEST_PREFIX rescue nil
```

File: src/etcd/lease.cr (hunk 1 above)
Comment (dukeraphaelng) [reply src/etcd/lease.cr:20]: Nope it's a pure Exception class atm. It should be a JSON parse exception though
https://github.com/crystal-lang/crystal/blob/cc22dc0682d69f9d79617b16af71d480a987adc0/src/json/pull_parser.cr#L385

**Pull request merged**